### PR TITLE
Add user profile picture in chat page

### PR DESCRIPTION
**Pull request by nganttran**

### CODE REVIEW

File: step/src/main/java/com/google/sps/MessageHandler.java [hunk 1]
```diff
@@ -65,9 +65,10 @@ public static void addMessage(Message m) {
             firstUserRecipientFilter, secondUserSenderFilter
     ))));
 
-    Query query = new Query("Message").setFilter(compositeFilter).addSort("timestamp", SortDirection.ASCENDING);
+    Query query = new Query("Message").setFilter(compositeFilter);
+    //query.addSort("timestamp", SortDirection.ASCENDING);
     PreparedQuery results = datastore.prepare(query);
-    List<Message> messages = new ArrayList<>();
+    ArrayList<Message> messages = new ArrayList<>();
```
Comment (adamsmlsn) [inline step/src/main/java/com/google/sps/MessageHandler.java:72]: Why did you choose to reduce the scope of the messages object? In general, it's better to have a higher scope so that the return object is more accessible.

File: step/src/main/java/com/google/sps/MessageHandler.java [hunk 2]
```diff
@@ -65,9 +65,10 @@ public static void addMessage(Message m) {
             firstUserRecipientFilter, secondUserSenderFilter
     ))));
 
-    Query query = new Query("Message").setFilter(compositeFilter).addSort("timestamp", SortDirection.ASCENDING);
+    Query query = new Query("Message").setFilter(compositeFilter);
+    //query.addSort("timestamp", SortDirection.ASCENDING);
```
Comment (adamsmlsn) [inline step/src/main/java/com/google/sps/MessageHandler.java]: Can you delete this commented code if it's no longer needed?

File: step/src/main/java/com/google/sps/servlets/ChatServlet.java [hunk 3]
```diff
@@ -83,6 +85,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     UserService userService = UserServiceFactory.getUserService();
     String senderID = userService.getCurrentUser().getUserId();
     String recipientID = request.getParameter("user"); 
+    if (recipientID  == null) doGet(request, response);
```
Comment (adamsmlsn) [inline step/src/main/java/com/google/sps/servlets/ChatServlet.java]: Can you put this in proper Java style? This can be hard to read.

File: step/src/main/java/com/google/sps/servlets/MatchingServlet.java [hunk 4]
```diff
@@ -66,8 +66,8 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     }
     //Send all the matches to the front-end
     request.setAttribute("matches", userMatches); 
-    //response.sendRedirect("profile.jsp");
-    request.getRequestDispatcher("chat.jsp").forward(request, response);
+    response.sendRedirect("profile.jsp");
+    //request.getRequestDispatcher("chat.jsp").forward(request, response);
```
Comment (adamsmlsn) [inline step/src/main/java/com/google/sps/servlets/MatchingServlet.java]: Can this comment be deleted?

File: step/src/main/webapp/chat.jsp [hunk 5]
```diff
@@ -37,6 +37,7 @@ limitations under the License.
     </head>
     <body>
         <nav>
+            <a id="profile-button" href="profile.jsp"> Profile </a>
```
Comment (adamsmlsn) [inline step/src/main/webapp/chat.jsp]: Can you put something like "My Profile" here? I think just "Profile" could get confusing while displaying other users

File: step/src/main/webapp/chat.jsp [hunk 6]
```diff
@@ -167,25 +177,8 @@ limitations under the License.
...
-                imageContainer.appendChild(userIcon);
-            }
-
-            async function grabBlobURL(){
-                const blobURL = await fetch("/image-upload").then((response) => {return response.text();});
-                const myForm = document.getElementById("image-form");
-                myForm.action = blobURL;
-            }
+            document.cookie = 'same-site-cookie=foo; SameSite=None; Secure';
+            document.cookie = 'cross-site-cookie=bar; SameSite=None; Secure';
```
Comment (adamsmlsn) [inline step/src/main/webapp/chat.jsp]: Is this needed? Foo and bar usually aren't names for permanent things

**Review by adamsmlsn — CHANGES_REQUESTED** [review]
Requesting a few changes but overall this is a good feature!

File: step/src/main/webapp/chat.jsp (hunk 6 above)
Comment (lailagamaleldin) [reply step/src/main/webapp/chat.jsp]: Also, what are we using cookies for?

File: step/src/main/java/com/google/sps/MessageHandler.java (hunk 1 above)
Comment (nganttran) [reply step/src/main/java/com/google/sps/MessageHandler.java:72]: I could not think of a reason why but when I changed it back it had errors in some test cases. I might try to look over to see when it started having the error

**Review by lailagamaleldin — APPROVED** [review]

File: step/src/main/java/com/google/sps/servlets/ChatServlet.java [hunk 7]
```diff
@@ -85,7 +84,6 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     UserService userService = UserServiceFactory.getUserService();
     String senderID = userService.getCurrentUser().getUserId();
     String recipientID = request.getParameter("user"); 
-    if (recipientID  == null) doGet(request, response);
```
Comment (adamsmlsn) [inline step/src/main/java/com/google/sps/servlets/ChatServlet.java]: Why did you delete this line? Seems like something you might've needed

**Review by adamsmlsn — APPROVED** [review]
